kitchen/modules/weights/src/lib.rs

```diff
@@ -0,0 +1,177 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// Transaction Weight Examples
+// https://crates.parity.io/sr_primitives/weights/index.html
+use support::{ensure, decl_module, decl_storage, dispatch::{Result, WeighData}};
+use runtime_primitives::weights::{ DispatchClass, Weight, ClassifyDispatch, SimpleDispatchInfo };
+
+pub trait Trait: system::Trait {}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as SimpleMap {
+		StoredValue get(stored_value): u32;
+	}
+}
+
+// A "scale" to weigh transactions. This scale can be used with any transactions that take a
+// single argument of type u32. The ultimate weight of the transaction is the / product of the
+// transaction parameter and the field of this struct.
+pub struct Linear(u32);
+
+// The actual weight calculation happens in the
+// impl WeighData block
+impl WeighData<(&u32,)> for Linear {
+	fn weigh_data(&self, (x,): (&u32,)) -> Weight {
+
+		// Use saturation so that an extremely large parameter value
+		// Does not cause overflow.
```

Comment:
if above comment is accepted, delete this line
```suggestion
		
```
I'm also happy with other phrasing that just reveals more about how overflow is handled with `saturating_mul`

Reply:
Lines 26 and 27 were supposed to be a single comment with line 27 being the end of the sentence (should uncapitalize the D). Doe sit make sense with that correction? Or would you like me to say more about saturation. I just wonder if it's a weird place to insert that discussion. Really safe math could have it's own whole section.

Reply:
Fair, I'm happy with your wording.